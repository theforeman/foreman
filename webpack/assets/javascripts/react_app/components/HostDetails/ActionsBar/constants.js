export const SUPPORTED_ERRORS = {
  host: __('Host'),
  templates: __('Templates'),
  proxies: __('Proxies'),
};
export const API_OPTIONS = { key: 'BUILD_REVIEW' };
