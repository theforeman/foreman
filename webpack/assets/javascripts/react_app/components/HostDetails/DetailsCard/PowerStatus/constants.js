import { translate as __ } from '../../../../common/I18n';

export const POWER_REQURST_KEY = 'HOST_TOGGLE_POWER';
export const POWER_REQUEST_OPTIONS = { key: POWER_REQURST_KEY };
export const BASE_POWER_STATES = { off: __('Off'), on: __('On') };
export const BMC_POWER_STATES = { soft: __('Reboot'), cycle: __('Reset') };
export const SUPPORTED_POWER_STATES = {
  ...BASE_POWER_STATES,
  ...BMC_POWER_STATES,
};
