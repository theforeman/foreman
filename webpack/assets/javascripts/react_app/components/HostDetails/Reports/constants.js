export const REPORT_API_OPTIONS = { key: 'HOST_DETAILS_RECENT_REPORTS' };
