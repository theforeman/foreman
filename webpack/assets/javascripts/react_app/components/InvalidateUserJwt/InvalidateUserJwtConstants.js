export const INVALIDATE_USER_JWT = 'INVALIDATE_USER_JWT';
