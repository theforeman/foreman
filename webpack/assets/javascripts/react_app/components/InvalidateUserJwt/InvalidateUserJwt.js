import React from 'react';
import PropTypes from 'prop-types';

import {
  Alert,
  Button,
  Grid,
  GridItem,
  Modal,
  ModalVariant,
} from '@patternfly/react-core';

import { translate as __ } from '../../common/I18n';
import { STATUS } from '../../constants';

const InvalidateUserJwt = ({
  isModalOpen,
  handleModal,
  handleSubmit,
  apiStatus,
  isLoading,
}) => (
  <>
    <Grid hasGutter>
      <GridItem span={12}>
        <Button
          onClick={() => handleModal(true)}
          variant="danger"
          isLoading={isLoading}
        >
          {isLoading ? __('Invalidating ...') : __('Invalidate JWTs')}
        </Button>
      </GridItem>
      <Modal
        variant={ModalVariant.small}
        title={__('Invalidate JSON web tokens')}
        isOpen={isModalOpen}
        onClose={() => handleModal(false)}
        actions={[
          <Button
            key="confirm"
            variant="primary"
            onClick={() => handleSubmit()}
          >
            {__('Confirm')}
          </Button>,
          <Button
            key="cancel"
            variant="link"
            onClick={() => handleModal(false)}
          >
            {__('Cancel')}
          </Button>,
        ]}
      />
      {apiStatus === STATUS.ERROR && (
        <GridItem span={8}>
          <Alert
            variant="danger"
            title={__(
              'JSON web tokens could not be invalidated, please see the log for more details'
            )}
          />
        </GridItem>
      )}
      {apiStatus === STATUS.RESOLVED && (
        <GridItem span={8}>
          <Alert
            variant="success"
            title={__('JSON web tokens successfully invalidated')}
          />
        </GridItem>
      )}
    </Grid>
  </>
);

InvalidateUserJwt.propTypes = {
  isModalOpen: PropTypes.bool.isRequired,
  handleModal: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  apiStatus: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
};

InvalidateUserJwt.defaultProps = {
  apiStatus: '',
};

export default InvalidateUserJwt;
