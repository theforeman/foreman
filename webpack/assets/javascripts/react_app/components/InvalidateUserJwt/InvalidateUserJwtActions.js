import { INVALIDATE_USER_JWT } from './InvalidateUserJwtConstants';
import { foremanUrl } from '../../../foreman_tools';
// import * as APIActions from '../../redux/API';
import { APIActions } from '../../redux/API';

export const invalidateJwtAction = id =>
  APIActions.delete({
    key: INVALIDATE_USER_JWT,
    url: foremanUrl(`/api/v2/users/${id}/invalidate_jwts`),
  });
