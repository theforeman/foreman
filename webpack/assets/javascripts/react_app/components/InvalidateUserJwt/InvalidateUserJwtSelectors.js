import { INVALIDATE_USER_JWT } from './InvalidateUserJwtConstants';
import { selectAPIStatus } from '../../redux/API/APISelectors';

export const selectStatus = state =>
  selectAPIStatus(state, INVALIDATE_USER_JWT);
