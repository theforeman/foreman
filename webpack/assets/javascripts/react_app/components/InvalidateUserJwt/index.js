import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useForemanUser } from '../../Root/Context/ForemanContext';
import { invalidateJwtAction } from './InvalidateUserJwtActions';
import { selectStatus } from './InvalidateUserJwtSelectors';
import { STATUS } from '../../constants';

import InvalidateUserJwt from './InvalidateUserJwt';

const InvalidateUserJwtWrapper = () => {
  const dispatch = useDispatch();
  const { id } = useForemanUser();
  const apiStatus = useSelector(selectStatus);

  const isLoading = apiStatus === STATUS.PENDING;
  const [isModalOpen, setIsModalOpen] = useState(false);

  const invalidateJWT = () => {
    setIsModalOpen(false);
    dispatch(invalidateJwtAction(id));
  };

  return (
    <InvalidateUserJwt
      isModalOpen={isModalOpen}
      handleModal={setIsModalOpen}
      handleSubmit={invalidateJWT}
      apiStatus={apiStatus}
      isLoading={isLoading}
    />
  );
};

export default InvalidateUserJwtWrapper;
