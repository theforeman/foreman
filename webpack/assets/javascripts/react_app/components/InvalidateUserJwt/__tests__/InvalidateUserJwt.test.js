import { STATUS } from '../../../constants';
import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';
import InvalidateUserJwt from '../InvalidateUserJwt';

const baseFixtures = {
  isModalOpen: false,
  handleModal: () => {},
  handleSubmit: () => {},
  apiStatus: undefined,
  isLoading: false
}
const fixtures = {
  'renders': baseFixtures,
  'pending': {...baseFixtures, apiStatus: STATUS.PENDING, isLoading: true},
  'success': {...baseFixtures, apiStatus: STATUS.RESOLVED},
  'error': {...baseFixtures, apiStatus: STATUS.ERROR},
};

describe('InvalidateUserJwt', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(InvalidateUserJwt, fixtures));
  });
