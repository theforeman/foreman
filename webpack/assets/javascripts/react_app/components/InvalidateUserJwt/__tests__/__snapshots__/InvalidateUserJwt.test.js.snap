// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidateUserJwt rendering error 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
  >
    <GridItem
      span={12}
    >
      <Button
        isLoading={false}
        onClick={[Function]}
        variant="danger"
      >
        Invalidate JWTs
      </Button>
    </GridItem>
    <Modal
      actions={
        Array [
          <Button
            onClick={[Function]}
            variant="primary"
          >
            Confirm
          </Button>,
          <Button
            onClick={[Function]}
            variant="link"
          >
            Cancel
          </Button>,
        ]
      }
      appendTo={[Function]}
      aria-describedby=""
      aria-label=""
      aria-labelledby=""
      className=""
      hasNoBodyWrapper={false}
      isOpen={false}
      onClose={[Function]}
      ouiaSafe={true}
      showClose={true}
      title="Invalidate JSON web tokens"
      titleIconVariant={null}
      titleLabel=""
      variant="small"
    />
    <GridItem
      span={8}
    >
      <Alert
        title="JSON web tokens could not be invalidated, please see the log for more details"
        variant="danger"
      />
    </GridItem>
  </Grid>
</Fragment>
`;

exports[`InvalidateUserJwt rendering pending 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
  >
    <GridItem
      span={12}
    >
      <Button
        isLoading={true}
        onClick={[Function]}
        variant="danger"
      >
        Invalidating ...
      </Button>
    </GridItem>
    <Modal
      actions={
        Array [
          <Button
            onClick={[Function]}
            variant="primary"
          >
            Confirm
          </Button>,
          <Button
            onClick={[Function]}
            variant="link"
          >
            Cancel
          </Button>,
        ]
      }
      appendTo={[Function]}
      aria-describedby=""
      aria-label=""
      aria-labelledby=""
      className=""
      hasNoBodyWrapper={false}
      isOpen={false}
      onClose={[Function]}
      ouiaSafe={true}
      showClose={true}
      title="Invalidate JSON web tokens"
      titleIconVariant={null}
      titleLabel=""
      variant="small"
    />
  </Grid>
</Fragment>
`;

exports[`InvalidateUserJwt rendering renders 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
  >
    <GridItem
      span={12}
    >
      <Button
        isLoading={false}
        onClick={[Function]}
        variant="danger"
      >
        Invalidate JWTs
      </Button>
    </GridItem>
    <Modal
      actions={
        Array [
          <Button
            onClick={[Function]}
            variant="primary"
          >
            Confirm
          </Button>,
          <Button
            onClick={[Function]}
            variant="link"
          >
            Cancel
          </Button>,
        ]
      }
      appendTo={[Function]}
      aria-describedby=""
      aria-label=""
      aria-labelledby=""
      className=""
      hasNoBodyWrapper={false}
      isOpen={false}
      onClose={[Function]}
      ouiaSafe={true}
      showClose={true}
      title="Invalidate JSON web tokens"
      titleIconVariant={null}
      titleLabel=""
      variant="small"
    />
  </Grid>
</Fragment>
`;

exports[`InvalidateUserJwt rendering success 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
  >
    <GridItem
      span={12}
    >
      <Button
        isLoading={false}
        onClick={[Function]}
        variant="danger"
      >
        Invalidate JWTs
      </Button>
    </GridItem>
    <Modal
      actions={
        Array [
          <Button
            onClick={[Function]}
            variant="primary"
          >
            Confirm
          </Button>,
          <Button
            onClick={[Function]}
            variant="link"
          >
            Cancel
          </Button>,
        ]
      }
      appendTo={[Function]}
      aria-describedby=""
      aria-label=""
      aria-labelledby=""
      className=""
      hasNoBodyWrapper={false}
      isOpen={false}
      onClose={[Function]}
      ouiaSafe={true}
      showClose={true}
      title="Invalidate JSON web tokens"
      titleIconVariant={null}
      titleLabel=""
      variant="small"
    />
    <GridItem
      span={8}
    >
      <Alert
        title="JSON web tokens successfully invalidated"
        variant="success"
      />
    </GridItem>
  </Grid>
</Fragment>
`;
